resourcemanager/identity/system_and_user_assigned_list.go

```diff
@@ -3,17 +3,18 @@ package identity
 import (
 	"encoding/json"
 	"fmt"
+
 	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
 var _ json.Marshaler = &SystemAndUserAssignedList{}
 
 type SystemAndUserAssignedList struct {
-	Type        Type     `json:"type"`
-	PrincipalId string   `json:"principalId"`
-	TenantId    string   `json:"tenantId"`
-	IdentityIds []string `json:"userAssignedIdentities"`
+	Type        Type     `json:"type" tfschema:"type"`
+	PrincipalId string   `json:"principalId" tfschema:"principal_id"`
+	TenantId    string   `json:"tenantId" tfschema:"tenant_id"`
+	IdentityIds []string `json:"userAssignedIdentities" tfschema:"identity_ids"`
```

Comment:
does this work? These are intentionally a Set rather than a List in the common schema, and iirc right now we only deserialize a List?

Reply:
I believe decoding [does work](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/sdk/resource_decode.go#L127-L129)? Although it doesn't have test coverage. I [gave it a try](https://github.com/hashicorp/terraform-provider-azurerm/commit/0d1c8f6532f097a6912f7eedce813435690c91c6) and it seems ok.

Encoding should also work, since it's only concerned with the values passed to d.Set() - in this case would be a slice regardless of whether the underlying type is a List or a Set?